test/jsonapi/serializer_test.exs

```diff
@@ -536,39 +535,31 @@ defmodule JSONAPI.SerializerTest do
     Application.delete_env(:jsonapi, :remove_links)
   end
 
-  test "serialize includes pagination links if they are defined and with_pagination is configured" do
-    data = %{id: 1}
-    Application.put_env(:jsonapi, :with_pagination, true)
+  test "serialize includes pagination links if page-based pagination is requested" do
+    data = [%{id: 1}]
 
-    encoded = Serializer.serialize(PostView, data, nil)
+    conn =
+      %Plug.Conn{}
```

Comment:
Do we want to use `conn/3` here from Plug.Test?

Reply:
See below.